File: Greenhouse-backend/src/main/java/dev/canercin/greenhouseiot/dto/GreenhouseDailyAverageDTO.java
package dev.canercin.greenhouseiot.dto;

import lombok.Data;

import java.sql.Date;

@Data
public class GreenhouseDailyAverageDTO {
    private Date date;
    private double avgHumidity;
    private double avgLight;
    private double avgSoilMoisture;
    private double avgTemperatureAsc;
    private double avgTemperatureAsf;
    private double avgWaterLevel;
}

File: Greenhouse-FrontEnd/src/app/models/GreenhouseDailyAverageDTO.ts
export class GreenhouseDailyAverageDTO{
  date: string;
  avgHumidity: number;
  avgLight: number;
  avgSoilMoisture: number;
  avgTemperatureAsc: number;
  avgTemperatureAsf: number;
  avgWaterLevel: number;

  constructor(date: string, avgHumidity: number, avgLight: number, avgSoilMoisture: number, avgTemperatureAsc: number, avgTemperatureAsf: number, avgWaterLevel: number) {
    this.date = date;
    this.avgHumidity = avgHumidity;
    this.avgLight = avgLight;
    this.avgSoilMoisture = avgSoilMoisture;
    this.avgTemperatureAsc = avgTemperatureAsc;
    this.avgTemperatureAsf = avgTemperatureAsf;
    this.avgWaterLevel = avgWaterLevel;
  }
}
